packages/medusa-plugin-auth/src/auth-strategies/cognito/startegy.ts

```diff
@@ -0,0 +1,184 @@
+// Load modules.
+var util = require('util');
+import passport from 'passport-strategy';
+import {
+	CognitoIdentityProviderClient,
+	GetUserCommand,
+	GetUserCommandOutput,
+	AdminInitiateAuthCommand,
+	NotAuthorizedException,
+	AdminInitiateAuthCommandOutput,
+} from '@aws-sdk/client-cognito-identity-provider';
+import { CognitoOptions } from './types';
+
+type GetUserByRefreshTokenOutput = {
+	newAccessToken: string;
+	user: GetUserCommandOutput;
+};
+
+/**
+ * This function uses the opensource [`passport-startegy`](https://www.npmjs.com/package/passport-strategy/) to create a custom strategy for Cognito.
+ *
+ * @param options: CognitoOption
+ * @param verify: Function
+ */
+function CognitoStrategy(options: CognitoOptions, verify) {
+	if (!options) throw new Error('Cognito strategy requires options');
+	if (!verify) throw new Error('Cognito strategy requires a verify callback');
+
+	passport.Strategy.call(this);
+	this.region = options.region;
+	this.name = options;
+	this._userPoolId = options.userPoolId;
+	this._clientId = options.clientId;
+	this._accessKeyId = options.accessKeyId;
+	this._secretAccessKey = options.secretAccessKey;
+	this._verify = verify;
+	this.cognitoClient = new CognitoIdentityProviderClient({
+		region: this.region,
+		apiVersion: '2016-04-18',
```

Comment:
Is this the latest version?

Reply:
Oh my bad. It's not actually necessary to specify the `apiVersion`.

https://github.com/awslabs/smithy-typescript/blob/b8635f0a9774ababfa6fc146b346c76945b74a2e/packages/smithy-client/src/client.ts#L7C44-L7C44